COMMO.GameServer/World/House.cs

```diff
@@ -3,22 +3,57 @@
 
 namespace COMMO.GameServer.World {
 
-    public sealed class House {
-        public readonly UInt32 Id;
-        public string HouseName;
-        public Position EntryPosition;
+	public sealed class House {
+		public readonly UInt32 Id;
+		public string HouseName;
+		public Coordinate EntryDoor;
 
-        public readonly List<HouseTile> Tiles = new List<HouseTile>();
+		public readonly List<Tile> Tiles = new List<Tile>();
 
-        public House(UInt32 id) {
-            Id = id;
-        }
+		public int LeftmostTileCoordinate { get; private set; }
+		public int TopmostTileCoordinate { get; private set; }
+		public int RightmostTileCoordinate { get; private set; }
+		public int BottommostTileCoordinate { get; private set; }
 
-        public void AddTile(HouseTile tile) {
-            if (tile == null)
-                throw new ArgumentNullException(nameof(tile));
+		public House(UInt32 id) {
+			Id = id;
+		}
 
-            Tiles.Add(tile);
-        }
-    }
+		public void AddTile(Tile tile) {
+			if (tile == null)
+				throw new ArgumentNullException(nameof(tile));
+
+			Tiles.Add(tile);
+
+			LeftmostTileCoordinate = Math.Min(LeftmostTileCoordinate, tile.Position.X);
+			TopmostTileCoordinate = Math.Min(TopmostTileCoordinate, tile.Position.Y);
+			RightmostTileCoordinate = Math.Max(RightmostTileCoordinate, tile.Position.X);
+			BottommostTileCoordinate = Math.Max(BottommostTileCoordinate, tile.Position.Y);
+		}
+
+		public bool TryGetTile(in Coordinate location, out Tile tile) {
+			if (location.X < LeftmostTileCoordinate ||
+				location.X > RightmostTileCoordinate ||
+				location.Y < TopmostTileCoordinate ||
+				location.Y > BottommostTileCoordinate) {
+
+				tile = null;
+				return false;
+			}
+
+			for (int i = 0; i < Tiles.Count; i++) {
```

Comment:
What about to rewrite that using System.Linq?

Reply:
Sadly Linq is not performant enough. This methods will be called many times per server tick. We can't even afford to use a foreach instead of a for.